Firestore/Source/Local/FSTRemoteDocumentChangeBuffer.mm

```diff
@@ -32,13 +37,12 @@ - (instancetype)initWithCache:(id<FSTRemoteDocumentCache>)cache;
 /** The underlying cache we're buffering changes for. */
 @property(nonatomic, strong, nonnull) id<FSTRemoteDocumentCache> remoteDocumentCache;
 
-/** The buffered changes, stored as a dictionary for easy lookups. */
-@property(nonatomic, strong, nullable)
-    NSMutableDictionary<FSTDocumentKey *, FSTMaybeDocument *> *changes;
-
 @end
 
-@implementation FSTRemoteDocumentChangeBuffer
+@implementation FSTRemoteDocumentChangeBuffer {
+  /** The buffered changes, stored as a dictionary for easy lookups. */
+  std::unique_ptr<std::map<DocumentKey, FSTMaybeDocument *>> _changes;
```

Comment:
Does this need to be a unique_ptr? Why not just a (non-pointer) map?

Update: Oh, I see. This is to allow `changes.reset(nullptr)` on line 84. Rather than dealing with pointers, I'd probably just add a bool to track the state. Optional.